include/mqtt/endpoint.hpp

```diff
@@ -1005,6 +1005,46 @@ class endpoint : public std::enable_shared_from_this<endpoint<Mutex, LockGuard,
         );
     }
 
+    /**
+     * @brief Publish with already acquired packet identifier
+     * @param packet_id
+     *        packet identifier. It should be acquired by acquire_unique_packet_id, or register_packet_id.
+     *        The ownership of  the packet_id moves to the library.
+     *        If qos == qos::at_most_once, packet_id must be 0. But not checked in release mode due to performance.
+     * @param topic_name
+     *        A topic name to publish
+     * @param contents
+     *        The contents to publish
+     * @param pubopts
+     *        qos, retain flag, and dup flag.
+     * @param life_keeper
+     *        An object that stays alive as long as the library holds a reference to any other parameters.
+     *        If topic_name, contents, or props do not have built-in lifetime management, (e.g. MQTT_NS::buffer)
+     *        use this parameter to manage their lifetime.
+     *
+     * @note If your QOS level is exactly_once or at_least_once, then the library will store this publish
+     *       internally until the broker has confirmed delivery, which may involve resends, and as such the
+     *       life_keeper parameter is important.
+     */
+    void publish(
+        packet_id_t packet_id,
+        as::const_buffer topic_name,
+        as::const_buffer contents,
+        publish_options pubopts = {},
+        any life_keeper = {}
+    ) {
+        BOOST_ASSERT((pubopts.get_qos() == qos::at_most_once && packet_id == 0) || (pubopts.get_qos() != qos::at_most_once && packet_id != 0));
+
+        send_publish(
+            packet_id,
+            topic_name,
+            contents,
+            pubopts,
+            v5::properties{},
+            force_move(life_keeper)
+        );
+    }
+
```

Comment:
Need overload without v5::properties = {} for std::string version, and MQTT_NS::buffer version, as well.

Reply:
I think that `std::string` version doesn't need the overload. Because topic_name and contents don't need the life_keeper. The life_keeper is for properties.

I think that `MQTT_NS::buffer` version is difficult.
If MQTT_NS::buffer owns lifetime, then life_keeper isn't needed.
If MQTT_NS::buffer doesn't have lifetime, then I personally think using `as::buffer` is better choice.
But mqtt_cpp accepts MQTT_NS::buffer that doesn't have its lifetime, so the overload should be added.